### PR TITLE
DIRECTOR: Fix normaliseXLibName for kPlatformWindows

### PR DESCRIPTION
**Pull request by moralrecordings**

### CODE REVIEW

File: engines/director/detection_tables.h [hunk 1]
```diff
@@ -1071,7 +1071,6 @@ static const PlainGameDescriptor directorGames[] = {
 	{ "fantasystudio",		"Fantasy Studio" }, // RPG Maker-style creation toolkit
 	{ "fototune",			"FotoTune Multimedia Show" },
 	{ "fukuda",				"Fukuda Shigeo Retrospective Collection" },
-	{ "fukuokagoround",		"Fukuoka Go-Round" },
```
Comment (mistydemeo) [inline engines/director/detection_tables.h:1074]: I don't think you meant to revert this part.

**Comment by mistydemeo** [comment]
Tested, and Fukuoka Go-Round is still able to access the main menu on both Mac and Windows on this branch.

File: engines/director/lingo/lingo-builtins.cpp [hunk 2]
```diff
@@ -2033,50 +2033,49 @@ void LB::b_installMenu(int nargs) {
...
+	// Continuation character is 0xac to denote a line running over.
+	// For Mac, this is ¨. For Windows, this is ¬.
+	const char CONTINUATION_CHAR = -84;
+	// Menu definitions use the character 0xc3 to denote a checkmark.
+	// For Mac, this is √. For Windows, this is Ã.
+	// This is used by MacMenu::createSubMenuFromString.
+
+	Common::String line;
+
+	for (auto it = menuStxt.begin(); it != menuStxt.end(); ++it) {
```
Comment (sev-) [inline engines/director/lingo/lingo-builtins.cpp:2054]: Just for the future, now we can write foreach: `for (auto &it : menuStxt)` which I consider cool.

File: engines/director/lingo/lingo-builtins.cpp [hunk 3]
```diff
@@ -2033,50 +2033,49 @@ void LB::b_installMenu(int nargs) {
...
-				s++;
-			} else if (*s == CODE_SEPARATOR_U32) {
-				lineBuffer += CODE_SEPARATOR_CHAR;
-				s++;
-			} else if (*s == CONTINUATION) { // fast forward to the next line
-				s++;
-				if (*s == LINE_BREAK) {
-					lineBuffer += ' ';
-					s++;
+	const char CODE_SEPARATOR_CHAR = -59;
```
Comment (sev-) [inline engines/director/lingo/lingo-builtins.cpp:2044]: This is problematic. Please use `byte` instead, which is unsigned. The thing is that `char` could be signed or unsigned depending on the platform and having a negative here could lead to portability problems.

The most portable and dev-friendly version is to use hex constants `0xC5` or the hex notation like `\xC5`

File: engines/director/lingo/lingo-object.cpp [hunk 4]
```diff
@@ -208,6 +208,8 @@ Common::String Lingo::normalizeXLibName(Common::String name) {
 		if (name.hasSuffixIgnoreCase(".xlib"))
 			name = name.substr(0, name.size() - 5);
 	} else if (platform == Common::kPlatformWindows) {
+		int pos = name.findLastOf("\\");
```
Comment (sev-) [inline engines/director/lingo/lingo-object.cpp:211]: This should be `size_t` to match the prototype of the `findLastOf`. Also, you need to check it against `Common::npos` when the separator is not found. You better not assume that `npos` currently equals `-1`, e.g. `pos + 1`

File: engines/director/resource.cpp [hunk 5]
```diff
@@ -193,48 +191,28 @@ void Window::probeMacBinary(MacArchive *archive) {
...
 
-Archive *Window::openMainArchive(const Common::String movie) {
-	debug(1, "openMainArchive(\"%s\")", movie.c_str());
+Archive *Window::openArchive(const Common::String movie) {
+	debug(1, "openArchive(\"%s\")", movie.c_str());
 
-	// If the archive is already open, don't reopen it;
-	// just init from the existing archive. This prevents errors that
-	// can happen when trying to load the same archive more than once.
-	if (g_director->_allOpenResFiles.contains(movie) && SearchMan.hasFile(movie)) {
```
Comment (sev-) [inline engines/director/resource.cpp:202]: Why was this removed?

File: engines/director/resource.cpp [hunk 6]
```diff
@@ -251,24 +229,33 @@ Archive *Window::loadEXE(const Common::String movie) {
 	} else {
 		warning("No LINGO.INI");
 	}
+}
 
+Archive *Window::loadEXE(const Common::String movie) {
 	Common::SeekableReadStream *exeStream = SearchMan.createReadStreamForMember(Common::Path(movie, g_director->_dirSeparator));
 	if (!exeStream)
-		return nullptr;
+		error("Failed to open EXE '%s'", g_director->getEXEName().c_str());
```
Comment (sev-) [inline engines/director/resource.cpp:237]: Please put the method name for easier debugging, e.g. `Window::loadEXE(): Failed to ope...` and the same with the error messages below.

**Review by sev- — CHANGES_REQUESTED** [review]
Reviewed and added my notes

File: engines/director/lingo/lingo-builtins.cpp (hunk 2 above)
Comment (moralrecordings) [reply engines/director/lingo/lingo-builtins.cpp:2054]: I don't think we can use that shortcut here, as that'll dereference it and we need to mess with the iterator inside the loop.

File: engines/director/lingo/lingo-builtins.cpp (hunk 3 above)
Comment (moralrecordings) [reply engines/director/lingo/lingo-builtins.cpp:2044]: Thanks, casting unsigned 0xc5 to signed raises compiler errors and 2am brain forgot about `\xC5'

File: engines/director/resource.cpp (hunk 6 above)
Comment (moralrecordings) [reply engines/director/resource.cpp]: Done

File: engines/director/lingo/lingo-object.cpp (hunk 4 above)
Comment (moralrecordings) [reply engines/director/lingo/lingo-object.cpp]: Fixed

**Comment by sev-** [comment]
Merging
